### DISCUSSION TITLE
How do I combine multiple Tailwind CSS classes into a single one?

### DISCUSSION THREAD

**Discussion by oubihis** [tailwindlabs/tailwindcss#2235] [Help] [answered]
What saddens me in TailwindCss framework, is a large number of Classes, is there a solution to make multiple classes in one single class?

Example: 

`<button class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">
  Button
</button>`

---TO---

`<button class="hover:bg-blue-700 one-class">
  Button
</button>`

This is just Ideas thinking

Thanks

**Comment by simonswiss** [comment, marked as answer]
Hey @oubihis 👋

What you're describing is indeed possible with the [`@apply` directive](https://tailwindcss.com/docs/functions-and-directives#apply)!

You can have a play with the upcoming, improved behaviour of `@apply` by [turning on the experimental `applyComplexClasses` flag on](https://github.com/tailwindlabs/tailwindcss/releases/tag/v1.7.0#use-apply-with-variants-and-other-complex-classes).

Finally, here's a short screencast that shows you how to use `@apply` in a project: https://tailwindcss.com/course/composing-utilities-with-apply

I hope this helps! 👍

**Reply by oubihis** [reply]
Yeh @apply is what i need.

Thanks @simonswiss

**Reply by ilyaskarim** [reply]
@simonswiss can we do configure tailwind to do the same behavior in classes?

**Reply by wongjn** [reply]
Yep! Use `@apply` in a class rule.

**Reply by ilyaskarim** [reply]
Yes I am doing the same I was thinking if tailwind can convert element classes into single class. I guess that's not possible.
